packages/checkout/widgets-lib/src/widgets/sale/hooks/useSignOrder.ts

```diff
@@ -337,42 +375,54 @@ export const useSignOrder = (input: SignOrderInput) => {
       return [];
     }
 
-    let successful = true;
-    const execTransactions: ExecutedTransaction[] = [];
-
     const transactions = await filterAllowedTransactions(
       signData.transactions,
       provider,
     );
 
-    for (const transaction of transactions) {
-      const {
-        tokenAddress: to,
-        rawData: data,
-        methodCall: method,
-        gasEstimate,
-      } = transaction;
-      // eslint-disable-next-line no-await-in-loop
-      const [hash, txnError] = await sendTransaction(
-        to,
-        data,
-        gasEstimate,
-        method,
-      );
-
-      if (txnError || !hash) {
-        successful = false;
-        onTxnError(txnError, execTransactions);
-        break;
+    if (!isManualExecution) {
+      let successful = true;
+      for (const transaction of transactions) {
+        // eslint-disable-next-line no-await-in-loop
+        const success = await executeTransaction(
+          transaction,
+          onTxnSuccess,
+          onTxnError,
+        );
+        if (!success) {
+          successful = false;
+          break;
+        }
       }
-
-      execTransactions.push({ method, hash });
-      onTxnSuccess({ method, hash });
+      (successful ? setExecuteDone : setExecuteFailed)();
+    } else {
+      setCurrentTransactionIndex(0);
     }
 
-    (successful ? setExecuteDone : setExecuteFailed)();
+    return executeResponse.transactions;
+  };
+
+  const executeNextTransaction = async (
```

Comment:
Is it worth making these functions `executeNextTransaction` and `execute` callbacks? with the required dependencies referenced, I'm thinking otherwise each render these functions are being redefined and can potentially trigger re-renders.